File: HomeWork/HomeWork10/HomeWork10.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def level_order(root):
    ret_list = []
    level_order_helper(root, ret_list, 0)
    return ret_list

def level_order_helper(curr, ret_list, level):
    if curr:
        if len(ret_list) <= level:
            ret_list.append([])
        ret_list[level].append(curr.val)
        level_order_helper(curr.left, ret_list, level + 1)
        level_order_helper(curr.right, ret_list, level + 1)

def main():
    root = TreeNode(4)
    root.left = TreeNode(3)
    root.left.left = TreeNode(1)
    root.right = TreeNode(8)
    root.right.left = TreeNode(5)
    root.right.right = TreeNode(9)

    print(level_order(root))

if __name__ == "__main__":
    main()

File: HomeWork/HomeWork10/HomeWork10.java
package HomeWork.HomeWork10;

import java.util.*;

public class HomeWork10{
    public static void main(String[] args){
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(3);
        root.left.left = new TreeNode(1);
        root.right = new TreeNode(8);
        root.right.left = new TreeNode(5);
        root.right.right = new TreeNode(9);
       
        System.out.println(levelOrder(root));
    }
    public static List<List<Integer>> levelOrder(TreeNode root) {
       // Your code here
       List<List<Integer>> retList = new ArrayList<List<Integer>>();
       
       levelOrderHelper(root, retList, 0);
        return retList;
    }
    public static void levelOrderHelper(TreeNode curr, List<List<Integer>> retList, int level){
        if(curr != null){
            if(retList.size() <= level){
                retList.add(new ArrayList<Integer>());
            }
            retList.get(level).add(curr.val);
            levelOrderHelper(curr.left, retList, level+1);
            levelOrderHelper(curr.right, retList, level+1);
        }
    }

     
}
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {

        this.val = val;

        this.left = left;

        this.right = right;

    }
}